docs/guides/latest/volumesnapshot/statefulset.md

```diff
@@ -0,0 +1,828 @@
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: standard
+parameters:
+  type: pd-standard
+provisioner: pd.csi.storage.gke.io
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+```
+The [volumeBindingMode](https://kubernetes.io/docs/concepts/storage/storage-classes/#volume-binding-mode) field controls when volume binding and dynamic provisioning should occur. Kubernetes allows `Immediate` and `WaitForFirstConsumer` modes for binding volumes. The `Immediate` mode indicates that volume binding and dynamic provisioning occurs once the PVC is created and `WaitForFirstConsumer` mode indicates that volume binding and provisioning does not occur until a pod is created that uses this PVC.
+
+Let's create the `StorageClass` we have shown above,
+
+```console
+$ kubectl apply -f ./docs/examples/volume-snapshot/storageclass.yaml
+storageclass.storage.k8s.io/standard created
+```
+
+We also need a `VolumeSnapshotClass`. We are going to use the following `VolumeSnapshotClass` for this tutorial,
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1alpha1
+kind: VolumeSnapshotClass
+metadata:
+  annotations:
+    snapshot.storage.kubernetes.io/is-default-class: "true"
+  name: default-snapshot-class
+snapshotter: pd.csi.storage.gke.io
+```
+
+Here,
+
+* `metadata.annotations` annotations are used to set default [volumeSnapshotClass](https://kubernetes.io/blog/2018/10/09/introducing-volume-snapshot-alpha-for-kubernetes/).
+* `snapshotter` field to point to the respective CSI driver that is responsible for taking snapshot. As we are using [GCE Persistent Disk CSI Driver](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver), we are going to use `pd.csi.storage.gke.io` in this field.
+
+Let's create the `volumeSnapshotClass` crd we have shown above,
+```console
+$ kubectl apply -f ./docs/examples/volume-snapshot/default-volumesnapshotclass.yaml
+volumesnapshotclass.snapshot.storage.k8s.io/default-snapshot-class created
+```
+
+To keep everything isolated, we are going to use a separate namespace called `demo` throughout this tutorial.
+
+```console
+$ kubectl create ns demo
+namespace/demo created
+```
+
+>Note: YAML files used in this tutorial are stored in [/docs/examples/volume-snapshot](/docs/examples/volume-snapshot) directory of [appscode/stash](https://github.com/stashed/stash) repository.
+
+### Take Volume Snapshot
+
+When you create a Statefulset, there is no need to create PVCs separately, because all replicas in Statefulset use different PVCs to store data. Kubenetes allows us to define a `volumeClaimTemplates` in Statefulset so that new PVC is created for each replica automatically. 
+We will take snapshot of those PVCs using Stash.
+
+#### Deploy Statefulset:
+
+Now, We will deploy a Statefulset. This Statefulset will automatically generate sample data (`$(POD_NAME).txt` file; `$(POD_NAME)` are resolved by the pod name of the statefulset) in `/source/data` directory.
```

Comment:
just use:

This Statefulset will automatically generate sample data in `/source/data` directory.